scc/cert/certificate.go

```diff
@@ -20,11 +20,19 @@ func NewCertificate[S Statement](subject S) Certificate[S] {
 	return Certificate[S]{subject: subject}
 }
 
+func NewCertificateWithSignature[S Statement](subject S, signature AggregatedSignature[S]) Certificate[S] {
+	return Certificate[S]{subject: subject, signature: signature}
+}
+
 // Subject returns the statement that is certified by this certificate.
 func (c *Certificate[S]) Subject() S {
 	return c.subject
 }
 
+func (c *Certificate[S]) Signature() AggregatedSignature[S] {
+	return c.signature
```

Comment:
Please document that the returned signature is not a copy and mutations will affect the certificate.

Reply:
The signature is indeed deep copied. I will add documentation and test to reflect this.